File: php/fib.php
<?php
function fib($n)
{
    return $n < 2 ? $n : fib($n-1) + fib($n-2);
}

function fibLinear($n, $prevFib = 0, $fib = 1)
{
    if ($n < 2) {
        return $n;
    }
    while (--$n) {
        list($prevFib, $fib) = array($fib, $fib + $prevFib);
    }
    return $fib;
}

function fibFormula($n)
{
    return round(((5 ** .5 + 1) / 2) ** $n / 5 ** .5);
}

function fibTailRecursive($n, $prevFib = 0, $fib = 1)
{
    return $n === 0 ? $prevFib : fibTailRecursive($n - 1, $fib, $prevFib + $fib);
}

function fibGenerator($n, $prevFib = 0, $fib = 1)
{
    while ($n-- >= 0) {
        yield $prevFib;
        list($prevFib, $fib) = array($fib, $fib + $prevFib);
    }
}
function fibGenerate($n)
{
    foreach (fibGenerator($n) as $fib) echo $fib.PHP_EOL;
}

$input = count($argv) > 1 ? $argv[1] : 29;
echo(fibLinear($input).PHP_EOL);

File: swift/fib.swift
#if os(macOS) || os(iOS)
import Darwin
#elseif os(Linux) || CYGWIN
import Glibc
#endif

func fib(_ n: Int) -> Int {
  return n < 2 ? n : fib(n - 1) + fib(n - 2)
}

func fibLinear(_ n: Int) -> Int {
  var i = n
  var prevFib = 0
  var fib = 1
  while i > 0 {
    (prevFib, fib) = (fib, fib + prevFib)
    i -= 1;
  }
  return prevFib
}

func fibFormula(_ n: Int) -> Int {
  return Int(round(pow((pow(5, 0.5) + 1) / 2, Float(n)) / pow(5, 0.5)))
}

func fibFormulaAlternate(_ n: Int) -> Int {
  let sqrt5 = sqrt(5)
  let phi = (sqrt5 + 1) / 2
  let binet = pow(phi, Double(n)) / sqrt5
  return Int(round(binet))
}

func fibTailRecursive(_ n: Int, _ prevFib: Int = 0, _ fib: Int = 1) -> Int {
  return n == 0 ? prevFib : fibTailRecursive(n - 1, fib, prevFib + fib)
}

var args = CommandLine.arguments
var input = args.count > 1 ? Int(args[1]) ?? 29 : 29
print(fibLinear(input))
